docs(release): add `v1.3.1` release notes (#917)
openfga/openfga: CHANGELOG.md

## [Unreleased]

## [1.3.1] - 2023-08-02

### Changed
* Default Check and ListObjects concurrency read limits ([#916](https://github.com/openfga/openfga/pull/916))

>   In our last release [v1.3.0](https://github.com/openfga/openfga/releases/tag/v1.3.0) we modified the default behavior of Check and ListObjects such that it limits/restricts the degree of concurrency that is allowed for a single request. This change was unintended. This release reverts the default behavior back to unbounded concurrency limits (the prior default). The change mostly affects those using OpenFGA as a library.

Reviewer: Should we point them to how to set the values and what was the previous default value in 1.2.0?

Author: The limits didn't exist in `v1.2.0`, but they did in `v1.3.0`. Did you mean `v1.3.0`?

Author: @adriantam I'd like to point them here https://github.com/openfga/openfga.dev/pull/489

Can we merge that first and then I'll update this to point them there?

## [Unreleased]

## [1.3.1] - 2023-08-02

### Added
* Count datastore queries involved in Check resolution ([#880](https://github.com/openfga/openfga/pull/880))

  OpenFGA request logs will now include a field `datastore_query_count` that shows how many queries were involved in a single Check resolution.

### Fixed

> * Avoid panic due to incorrect error handling in Check ([#923](https://github.com/openfga/openfga/pull/923))

Reviewer 2: We don't need to mention this. This bug was introduced in https://github.com/openfga/openfga/pull/880 which we are including in this release

Author: True

Author: Dropped.

## [Unreleased]

## [1.3.1] - 2023-08-23

### Added

> * Count datastore queries involved in Check resolution ([#880](https://github.com/openfga/openfga/pull/880))

Reviewer: ```suggestion
* Count datastore queries involved in Check resolution metadata ([#880](https://github.com/openfga/openfga/pull/880))
```

Reviewer 2: > Count datastore queries involved in Check 

or

> Count datastore queries involved in Check and log it as resolution metadata

>   OpenFGA request logs will now include a field `datastore_query_count` that shows how many queries were involved in a single Check resolution.

Reviewer 3: They also now include the user agent

* Histogram metric to report the `datastore_query_count` per Check ([#924](https://github.com/openfga/openfga/pull/932))

  This new metric can be used to report percentiles of the number of database queries required to resolve Check requests.

> 

Author: Added

> * Prune user lookup optimization in Check ([#932](https://github.com/openfga/openfga/pull/932)

Reviewer 2: ```suggestion
* Prune user lookup optimization in Check ([#932](https://github.com/openfga/openfga/pull/932))
```

* CachedCheckResolver for caching Check subproblems ([#891](https://github.com/openfga/openfga/pull/891))

  This experimental feature adds new caching capabilities to the OpenFGA server. It is an "opt-in" feature and thus must be enabled. To enable this feature you must specify the experimental flag `check-query-cache` and set the  `--check-query-cache-enabled=true` flag.

  ```shell
  openfga run --experimentals check-query-cache --check-query-cache-enabled=true
  ```

### Changed
* Default Check and ListObjects concurrency read limits ([#916](https://github.com/openfga/openfga/pull/916))

  In our last release [v1.3.0](https://github.com/openfga/openfga/releases/tag/v1.3.0) we modified the default behavior of Check and ListObjects such that it limits/restricts the degree of concurrency that is allowed for a single request. This change was unintended. This release reverts the default behavior back to unbounded concurrency limits (the prior default). The change mostly affects those using OpenFGA as a library.

> 

Reviewer: Should call out community contribution for https://github.com/openfga/openfga/pull/941 and https://github.com/openfga/openfga/pull/927 in addition of bumping to Go 1.21

## [Unreleased]

## [1.3.1] - 2023-08-23

### Added
* Count datastore queries involved in Check resolution metadata ([#880](https://github.com/openfga/openfga/pull/880))

>   OpenFGA request logs will now include a field `datastore_query_count` that shows how many queries were involved in a single Check resolution.

Reviewer 2: It's not just logs right? It's also traces and metrics?

Reviewer 2: Ah it's below LOL

Author: We don't have a `datastore_query_count` label per metric for Check, those are bucketed. But we do have it in traces as well.

Author: Updated

* Histogram metric to report the `datastore_query_count` per Check ([#924](https://github.com/openfga/openfga/pull/932))

  This new metric can be used to report percentiles of the number of database queries required to resolve Check requests.

* Check request duration histogram labeled by method and datastore query count ([#950](https://github.com/openfga/openfga/pull/950))

  The `request_duration_by_query_count_ms` metric reports the total request duration (in ms) labelled by the RPC method and ranges of observations for the `datastore_query_count`. This metrics allows operators of an OpenFGA server to report request duration percentiles for Check requests based on the number of database queries that were required to resolve the query.

> * Prune user lookup optimization in Check ([#932](https://github.com/openfga/openfga/pull/932))

Reviewer 2: ```suggestion
* Optimize Check to avoid database lookups in some scenarios ([#932](https://github.com/openfga/openfga/pull/932))
```
